Documentation/ceph-csi-drivers.md

```diff
@@ -124,5 +124,152 @@ when pod get spawned and destroyed at time of pod delete.
+CR that includes the reclaim space as an operation.
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1alpha1
+kind: ReclaimSpaceJob
+metadata:
+  name: sample-1
+spec:
+  target:
+    persistentVolumeClaim: pvc-1
+  backOffLimit: 10
+  retryDeadlineSeconds: 900
+```
+
+A user that created the ReclaimSpaceJob can follow the progress, and
+check the result of the operation:
+
+```bash
+kubectl apply -f reclaimspace.yaml
+job.csiaddons.openshift.io/sample-1 created
+kubectl get reclaimspacejobs
+NAME                       COMPLETIONS   DURATION   AGE   SPACE RECLAIMED
+sample-1               0/1           6s         6s    -
+[admin@ocs]$ kubectl get reclaimspacejobs
+NAME                       COMPLETIONS   DURATION   AGE   SPACE RECLAIMED
+sample-1               1/1           10s        25s   153mb
+```
+
+### ReclaimSpaceCronJob
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1alpha1
+kind: ReclaimSpaceCronJob
+metadata:
+  name: reclaimspacecronjob-sample
+spec:
+  concurrencyPolicy: Forbid
+  failedJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      backOffLimit: 6
+      retryDeadlineSeconds: 600
+      target:
+        persistentVolumeClaim: data-pvc
+  schedule: '@weekly'
+  successfulJobsHistoryLimit: 3
+```
+
+### Annotating PerstentVolumeClaims
+
+`ReclaimSpaceCronJob` CR can also be automatically created by adding
+`reclaimspace.csiaddons.openshift.io/schedule: "@midnight"` annotation
+to PersistentVolumeClaim object.
+
+```bash
+$ kubectl get pvc data-pvc
+NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+data-pvc  Bound    pvc-f37b8582-4b04-4676-88dd-e1b95c6abf74   1Gi        RWO            default           20h
+
+$ kubectl annotate pvc data-pvc "reclaimspace.csiaddons.openshift.io/schedule=@midnight"
+persistentvolumeclaim/data-pvc annotated
+
+$ kubectl get reclaimspacecronjobs.csiaddons.openshift.io 
+NAME                    SCHEDULE    SUSPEND   ACTIVE   LASTSCHEDULE   AGE
+data-pvc-1642663516   @midnight                                     3s
+
+$ kubectl annotate pvc data-pvc "reclaimspace.csiaddons.openshift.io/schedule=*/1 * * * *" --overwrite=true
+persistentvolumeclaim/data-pvc annotated
+
+$ kubectl get reclaimspacecronjobs.csiaddons.openshift.io
+NAME                  SCHEDULE    SUSPEND   ACTIVE   LASTSCHEDULE   AGE
+data-pvc-1642664617   */1 * * * *                                   3s
+```
+
+* Upon adding annotation as shown above, a `ReclaimSpaceCronJob` with name
+  `"<pvc-name>-xxxxxxx"` is created (pvc name suffixed with current time hash when 
+  the job was created). 
+* `schedule` value is in the same [format as Kubernetes CronJobs][batch_cronjob]
+  that sets the and/or interval of the recurring operation request.
+* Default schedule value `"@weekly"` is used if `schedule` value is empty or in invalid format. 
+* `ReclaimSpaceCronJob` is recreated when `schedule` is modified and deleted when
+  the annotation is removed.
+
 ### Prerequisites
```

Comment:
Are these the prereqs for the csi addons, or for the previous section? If each section has a different version requirement, how about we include at the top of each section instead of at the end?

Reply:
not for csi-addons, will update it.